### DISCUSSION TITLE
Using release branches for major versions

### DISCUSSION THREAD

**Discussion by kmlebedev** [seaweedfs/seaweedfs#2795] [Ideas]
Details on the link 
https://docs.gitlab.com/ee/topics/gitlab_flow.html#release-branches-with-gitlab-flow

**Comment by chrislusf** [comment]
One master branch is conceptually easier though. Release branches will open up a box of more troubles, i.g., build process, back porting fixes, compatibility, reproducing bugs, ...

**Reply by kmlebedev** [reply]
> One master branch is conceptually easier though. Release branches will open up a box of more troubles, i.g., build process, back porting fixes, compatibility, reproducing bugs, ...

I know, but easier doesn't mean better. 
But you can at least try and limit yourself to one active branch with the major release version

**Reply by kmlebedev** [reply]
Also, now there is an understanding of which changes are major and which are minor.
Let me remind you that majors lead to a loss of backward compatibility and I would like to understand in which versions this happens

**Reply by chrislusf** [reply]
"loss of backward compatibility" this hopefully should never happen in the future. It happened once around 0.67 version.

**Reply by kmlebedev** [reply]
For me, this is a generally accepted deployment process in testing => pre-production => production
And for a new version to go to production, it takes from one to three weeks, depending on the bugs found.
Accordingly, I create this git process with release branches locally. Since in prod I want to get this version, which was in testing, but without bugs found

**Reply by kmlebedev** [reply]
> "loss of backward compatibility" this hopefully should never happen in the future. It happened once around 0.67 version.

backward compatibility in a broad sense is the different behavior of the application and the same launch parameters / configs
The latest prime example https://github.com/chrislusf/seaweedfs/issues/1937 and https://github.com/chrislusf/seaweedfs/commit/f247cab5cd47b7f54a3267979508160ebdd933ce
Two versions in a row is a strike

**Reply by kmlebedev** [reply]
@chrislusf UP https://github.com/chrislusf/seaweedfs/issues/3177
